query/src/main/java/org/infinispan/query/indexmanager/ClusteredSwitchingBackend.java

```diff
@@ -146,13 +146,18 @@ public synchronized void lazyInitialize() {
    }
 
    private synchronized void applyViewChangedEvent(ViewChangedEvent e) {
-      assert e != null;
-      assert e.getNewMembers().size() > 0;
-      assert e.getNewMembers().get(0) != null;
+      List<Address> newMembers = e.getNewMembers();
       if (log.isDebugEnabled()) {
-         log.debug("Notified of new View! Members: " + e.getNewMembers());
+         log.debug("Notified of new View! Members: " + newMembers);
       }
-      final Address newmaster = e.getNewMembers().get(0);
+      handleTopologyChange(newMembers);
+   }
+
+   private synchronized void handleTopologyChange(List<Address> newMembers) {
+      assert newMembers != null;
```

Comment:
Do we need those asserts? Testing code should to to `src/test` and nobody uses `-ea` on production. I would remove them.